src/rpc/server.cpp

```diff
@@ -424,8 +425,25 @@ static inline JSONRPCRequest transformNamedArguments(const JSONRPCRequest& in, c
             argsIn.erase(fr);
         } else {
             hole += 1;
+            if (out.params.empty()) initial_hole_size = hole;
         }
     }
+    // If leftover "args" param was found, use it as a source of positional
+    // arguments and add named arguments after. This is a convenience for
+    // clients that want to pass a combination of named and positional
+    // arguments, described in doc/JSON-RPC-interface.md#parameter-passing
+    auto positional_args = argsIn.find("args");
```

Comment:
could use [`extract`](https://en.cppreference.com/w/cpp/container/unordered_map/extract) here too, to avoid the need for `erase` later on?